athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/EdgeRowWriter.java

```diff
@@ -76,34 +82,34 @@ public static void writeRowTemplate(RowWriterBuilder rowWriterBuilder, Field fie
             case VARCHAR:
                 rowWriterBuilder.withExtractor(field.getName(),
                         (VarCharExtractor) (Object context, NullableVarCharHolder value) -> {
-                            Map<Object, Object> obj = (Map<Object, Object>) context;
+                            Map<String, Object> obj = (Map<String, Object>) contextAsMap(context, enableCaseinsensitivematch);
+                            String fieldName = field.getName();
                             value.isSet = 0;
-                            String fieldName = field.getName().toLowerCase().trim();
-                            
+
                             // check for special keys and parse them separately
                             if (fieldName.equals(SpecialKeys.ID.toString().toLowerCase())) {
-                                Object fieldValue = obj.get(T.id);
+                                Object fieldValue = obj.get(SpecialKeys.ID.toString());
                                 if (fieldValue != null) {
                                     value.value = fieldValue.toString();
                                     value.isSet = 1;
                                 }
-                            }
+                            } 
                             else if (fieldName.equals(SpecialKeys.IN.toString().toLowerCase())) {
-                                Object fieldValue = ((LinkedHashMap) obj.get(Direction.IN)).get(T.id);
+                                Object fieldValue = obj.get(SpecialKeys.IN.toString());
                                 if (fieldValue != null) {
                                     value.value = fieldValue.toString();
                                     value.isSet = 1;
                                 }
-                            }
+                            } 
                             else if (fieldName.equals(SpecialKeys.OUT.toString().toLowerCase())) {
-                                Object fieldValue = ((LinkedHashMap) obj.get(Direction.OUT)).get(T.id);
+                                Object fieldValue = obj.get(SpecialKeys.OUT.toString());
                                 if (fieldValue != null) {
                                     value.value = fieldValue.toString();
                                     value.isSet = 1;
                                 }
                             } 
                             else {
```

Comment:
It looks like all of these `if` and `else if` statements can be deleted now since they all just do the same thing as the `else` case (lines 112-116), right?

Reply:
Yes, seems like so, let me make changes and test

Reply:
Actually one thing you may want to do is perform the flattening in `contextAsMap` even when `caseInsensitive` is false to maintain the correct behavior regardless of casing.

Reply:
updated

Reply:
No, I mean that the deleted `if else` statements were basically doing an ad-hoc flattening of the map.

This isn't being done anymore when `caseInsensitive` is set to `false` (it doesn't matter what the default is).

So in order to maintain the same behavior as these deleted `if else` statements, regardless of the value of `caseInsensitive`, `contextAsMap` should always "flatten" out the map.

Reply:
Additionally, did you test to make sure that when `caseInsensitive` is set to false, that everything works as expected?

Reply:
thanks for pointing this, refactored code to make sure contextMap rewrite is consistent whether flag is set to true or false